deployment/terraform-django/database.tf

```diff
@@ -4,10 +4,10 @@
 resource "aws_db_subnet_group" "default" {
   name        = var.rds_database_identifier
   description = "Private subnets for the RDS instances"
-  subnet_ids  = aws_subnet.private.*.id
+  subnet_ids  = aws_subnet.public.*.id
```

Comment:
That's similar to what I originally tried:
LB/ECS on public subnet
ECS/RDS on private subnet

I couldn't get the terraform to with having ECS on both.

I just don't see the benefit to us for the increased complexity. With AWS public subnets we get effectively the ability to say on an instance by instance based if we want it to have a public IP or not.  Otherwise they only get private IPs. (Effectively we get two subnets with a public subnet)

Reply:
Oh, failed to test this because, without public IP it can't access ECR to pull the image it needs.  With the public IP it works.

I did find an alternative that would let us get away from needing public IPs for the containers - Egress only internet gateways using IPv6- https://docs.aws.amazon.com/vpc/latest/userguide/egress-only-internet-gateway.html.  That's likely the best long term option - but I don't think that's worth pursuing for echo.

Reply:
Cool, yeah, everything is clear now. My concluding high-level thoughts are that:
1. This is a good option for reducing costs on low-traffic, low-resource sites like this. NAT Gateways shouldn't double our hosting costs.
2. Even for low-traffic sites we may get client pushback if "security" is a concern because of the requirement for public IPs in this solution.
3. This is maybe another indication that a different provider like Digital Ocean or Heroku might be more appropriate for these types of sites?
4. I really like the sound of the IPv6 egress-only gateways and hope we can find a way to make that work on future projects because it sounds like it combines the best of both solutions (that is, it achieves low cost without requiring us to allocate static public IP addresses).